style: Fix all flake8-gettext (INT) errors (INT001, INT002, INT003)

echoix opened the pull request:
This fixes all instances in python files where translated strings are formatted before being fetched by gettext. (underscore function)
Ruff rules:
https://docs.astral.sh/ruff/rules/f-string-in-get-text-func-call/
https://docs.astral.sh/ruff/rules/format-in-get-text-func-call/
https://docs.astral.sh/ruff/rules/printf-in-get-text-func-call/

Some places I changed from percent formatting to dot format in order to use named parameters, to add more context for translators. Unfortunately, it might cause the source string to be different, and I'm not sure of the consequences on Weblate. From this FAQ https://docs.weblate.org/en/latest/faq.html#how-do-i-translate-several-branches-at-once, I understand that they don't really support translating for different branches, so having a source string changed makes it unavailable for backporting (if ever we are doing this). The translation files should be regenerated after this, as they are even out of date currently (merge conflicts when weblate tries to rebase because we use squash merge).

In a small number of cases, where I had added named format parameters, I also changed the strings around so they match the same string, and can be translated as one.

In `r.in.wms`, I'm pretty sure the variable "format" doesn't exist, and this, since the first commit of that file 12 years ago when it was migrated from an addon. I supposed it meant `self.gdal_drv_format`, from the context and another translated string near, but it is also possible it meant one of the parameters of the module (that wasn't called correctly if it was the case). Please take a look to confirm my guess.

These were all manual fixes, as no auto-fix was available.

I was committing in batches (also to help reviewing if it's too long in one go), I was planning on having the squash message clean this up.

echoix commented:
@neteler Maybe you have an opinion on this too

neteler commented:
> Unfortunately, it might cause the source string to be different, and I'm not sure of the consequences on Weblate.

I could imagine that all related existing translations would be changed to "fuzzy". If so, that might be very demotivating for the translators.

What does @HuidaeCho think?

echoix commented:
If the translation files are up to date before this PR, there should be a way to undo changes to the source strings, recreate the translation files (I expect no changes), apply back the source string changes (possibly in small batches), recreate the translation files, and manually/with a tool combine back the changes to the translations.

HuidaeCho commented:
> > Unfortunately, it might cause the source string to be different, and I'm not sure of the consequences on Weblate.
> 
> I could imagine that all related existing translations would be changed to "fuzzy". If so, that might be very demotivating for the translators.
> 
> What does @HuidaeCho think?

I'm not sure about that. These strings are still the same, so why would they become fuzzy?

echoix commented:
> > > Unfortunately, it might cause the source string to be different, and I'm not sure of the consequences on Weblate.
> 
> > 
> 
> > I could imagine that all related existing translations would be changed to "fuzzy". If so, that might be very demotivating for the translators.
> 
> > 
> 
> > What does @HuidaeCho think?
> 
> 
> 
> I'm not sure about that. These strings are still the same, so why would they become fuzzy? Because of line number changes? I'll try it when I get a chance.

That's the problem, some were changed just a little (placeholders), but maybe I could revert them.

echoix commented:
Also, if you get emails about the status updates of our weblate, it got relocked this morning after a new PR merged...

echoix commented on python/grass/temporal/univar_statistics.py:
```diff
@@ -65,11 +65,10 @@ def compute_univar_stats(registered_map_info, stats_module, fs, rast_region=Fals
-                "Unable to get statistics for {voxel}raster map "
-                "<{rmap}>".format(
-                    rmap=id, voxel="" if stats_module.name == "r.univar" else "3d "
-                )
+            _("Unable to get statistics for raster map <{rmap}>").format(rmap=id)
+            if stats_module.name == "r.univar"
+            else _("Unable to get statistics for 3d raster map <{rmap}>").format(
+                rmap=id
             )
         )
```
The source string doesn't appear in any .pot file in the repo, and can't be found in any of the 4 versions of grasslibs.pot (since april 2022) https://github.com/OSGeo/grass/commits/main/locale/templates/grasslibs.pot

However, when searching for "Unable to get statistics for" in the repo, we see that there are 3 strings that were already translated in grasslibs_ar.po, grasslibs_es.po, grasslibs_fr.po, grasslibs_pt_BR.po, grasslibs_ru.po, and grasslibs_sl.po (6 languages). Some are marked fuzzy, some don't.

If we take for example the french file, the strings already don't match the original strings here.

https://github.com/OSGeo/grass/blob/972bd32a23156f3255025a1ed6d42c9e563960b8/locale/po/grasslibs_fr.po#L9645-L9663

```
#, python-format
#~ msgid "Unable to get statistics for raster map <%s>"
#~ msgstr "Impossible d'obtenir les statistiques de la carte raster <%s>"

#, python-format
#~ msgid "Unable to get statistics for 3d raster map <%s>"
#~ msgstr "Impossible d'obtenir les statistiques pour la carte raster 3d <%s>"

#, python-format
#~ msgid "Space time %(sp)s dataset <%(i)s> not found"
#~ msgstr "Le jeu de données <%(i)s> temporel %(sp)s n'a pas été trouvé"

#, python-format
#~ msgid "Space time %(sp)s dataset <%(i)s> is empty"
#~ msgstr "Le jeu de données <%(i)s> temporel %(sp)s est vide"

#, python-format
#~ msgid "Unable to get statistics for vector map <%s>"
#~ msgstr "Impossible d'obtenir les statistiques pour la carte vecteur <%s>"
```



So, probably to make the existing translation match up again, it should be changed back to

```suggestion
    if not univar_stats:
        gs.warning(
            _("Unable to get statistics for raster map <%s>") % id
            if stats_module.name == "r.univar"
            else _("Unable to get statistics for 3d raster map <%s>") % id
        )
```


However, what I suggested for my PR makes it equivalent to the what was before this PR, if the intented formatting of `{voxel}` was applied before getting the translated string, but `{rmap}` was kept as a placeholder. But even in that case, the string doesn't exist in the .pot file.

wenzeslaus replied:
The original code is handling the strings in a way which assumes you can construct a sentence using string operations. This may work for some languages but not others and it seems to me as impossible to decipher when translating the strings by themselves. Having two complete sentences is the best practice here.

echoix replied:
I applied the change that will allow to match up existing translations to that code

echoix commented on gui/wxpython/gui_core/pyedit.py:
```diff
@@ -333,20 +333,18 @@ def _writeFile(self, file_path, content, additional_err_message=""):
                 message=_(
                     "Permission denied <{}>. Please change file "
-                    "permission for writing.{}".format(
-                        file_path,
-                        additional_err_message,
-                    ),
+                    "permission for writing.{}"
+                ).format(
+                    file_path,
+                    additional_err_message,
                 ),
```
Not having numbered or named placeholders makes it impossible in translations to reorder the placeholders to accommodate the different languages. My initial suggestion in this PR was adressing that

wenzeslaus replied:
I agree. Maybe this is for a different PR. There we can more precisely evaluate the impact of that particular change to existing translations.

echoix commented on gui/wxpython/wxgui.py:
```diff
@@ -100,10 +100,10 @@ def show_main_gui():
                         _(
                             "Current version of wxPython {} is lower than "
-                            "minimum required version {}".format(
-                                wx.__version__,
-                                ".".join(map(str, min_required_wx_version)),
-                            )
+                            "minimum required version {}"
+                        ).format(
+                            wx.__version__,
+                            ".".join(map(str, min_required_wx_version)),
                         )
                     )
                 else:
```
Here too, multiple placeholders that aren't numbered or named, so maybe it is impossible to translate correctly the string because the placeholders can't be reordered for the a specific language.

wenzeslaus replied:
This is the case also for several places with % formatting and `%s`. These will need to be changed at some point.

echoix replied:
For later

echoix commented:
I reverted the changes in the source strings in https://github.com/OSGeo/grass/pull/4052/commits/a39178cdba6557cd81fc38633582088d4967afd6, that ideally I would've kept. Its always possible to revert that commit in the PR.

I also added comments on some special cases, or cases where personally I disagree with the source string, and thus reverting to that string.

echoix commented on gui/wxpython/gui_core/pyedit.py:
```diff
@@ -508,9 +506,8 @@ def OpenRecentFile(self, path, file_exists, file_history):
-                    "File <{}> doesn't exist."
-                    "It was probably moved or deleted.".format(path)
+                _("File <{}> doesn't exist. It was probably moved or deleted.").format(
+                    path
```
The original string was missing a space between the two sentences. This is one source string change that I think we should keep.

wenzeslaus replied:
Please keep the change.

echoix commented:
@neteler and others. I'd like for this to be cleared during this week.

wenzeslaus commented:
> If so, that might be very demotivating for the translators.

Well, the current issue is that the messages are not translated at all in the application because of how the code is written (which is what this PR is fixing).

wenzeslaus commented on utils/mkhtml.py:
```diff
@@ -164,23 +164,22 @@ def download_git_commit(url, response_format, *args, **kwargs):
-                    "{code}, {desc}".format(
-                        url=url,
-                        code=response.code,
-                        desc=desc,
-                    ),
+                    "Download commit from <{url}>, return status code {code}, {desc}"
+                ).format(
+                    url=url,
+                    code=response.code,
+                    desc=desc,
                 ),
             )
```
Off topic: Should this be even translatable? It is compilation. Perhaps called though _g.extension_? But still, is this something people should spend their time with translating to other languages?

wenzeslaus commented on python/grass/temporal/univar_statistics.py:
```diff
@@ -65,11 +65,10 @@ def compute_univar_stats(registered_map_info, stats_module, fs, rast_region=Fals
-                )
+            _("Unable to get statistics for raster map <{rmap}>").format(rmap=id)
+            if stats_module.name == "r.univar"
+            else _("Unable to get statistics for 3d raster map <{rmap}>").format(
```
```suggestion
            else _("Unable to get statistics for 3D raster map <{rmap}>").format(
```

wenzeslaus replied:
And yes, the original string is wrong, so please keep you changes.

echoix replied:
I'll need to be on a computer to properly check what changes of the two suggestions to keep.

wenzeslaus reviewed (changes requested):
Looks good to me as it is except what is in the comments.

Just a heads up that keeping the messages same may have a similar frustration effect as suggested in a comment by @neteler just some time later. For example, string "Unable to remove timestamp for raster map" is correctly extracted for translation and if I understand correctly will therefore remain the same even after this PR. Because the string is no finally passed to the translate function (and not its modified version), it will be translated for the user for the first time only after this PR. Later, we may be changing all %-style strings or, much more likely, at least all strings with multiple `%s` or `{}` to format-style with keywords (`{name}`), so this will change the strings for sure. A string translated in the interface (thanks to this PR) will then be removed. So, same additional work for the translators as if we would change the strings now, but with more frustration with the final product where the once translated strings will disappear.

Overall, this PR (both the fixes and the enabled warning) is crucial for making the translation actually work consistently. The exact order of calls is clearly hard to grasp for contributors who don't deal with the translations themselves as demonstrated by the number of corrections needed here. This PR is fixing previous mistakes and avoids future ones.

wenzeslaus commented on gui/wxpython/gmodeler/panels.py:
```diff
@@ -1643,9 +1639,7 @@ def _layout(self):
-                parent=self, id=wx.ID_ANY, label="%s:" % _("Python script type")
-            ),
+            StaticText(parent=self, id=wx.ID_ANY, label=_("Python script type:")),
             flag=wx.ALIGN_CENTER_VERTICAL,
```
Yes, `"%s:" % _("Python script type")` is wrong.

wenzeslaus commented on gui/wxpython/gui_core/pyedit.py:
```diff
@@ -509,9 +507,8 @@ def OpenRecentFile(self, path, file_exists, file_history):
+                    "File <{path}> doesn't exist. It was probably moved or deleted."
+                ).format(path=path),
                 parent=self.guiparent,
             )
```
Change it and yes, I think update to use the keywords can be included as well.

echoix replied:
I changed to keep the path keyword too

echoix commented on gui/wxpython/iscatt/iscatt_core.py:
```diff
@@ -295,7 +295,7 @@ def _rasterize(self, grass_region, layer, cat, out_rast):
         if ret != 0:
-            GException(_("v.build failed:\n%s" % msg))
+            GException(_("v.build failed:\n{messages}").format(messages=msg))
 
```
I think there's a ruff rule or maybe a new Pylint warning with this category of errors, so it shouldn't be missed

wenzeslaus replied:
Definitively a different PR. And yes, I think I have seen a warning for this.

echoix commented on gui/wxpython/gmodeler/model.py:
```diff
@@ -2830,12 +2830,11 @@ def _getPythonActionCmd(self, item, task, cmdIndent, variables={}):
-                    f"not being written in the generated json: "
-                    f"{itemParameterizedFlags}"
-                ),
+                    "Module {task_name} in your model contains "
+                    "parameterized flags. actinia does not support "
+                    "parameterized flags. The following flags are therefore "
+                    "not being written in the generated json: {flags}"
+                ).format(task_name=task.get_name(), flags=itemParameterizedFlags),
                 caption=_("Warning"),
```
Yes, I remember renaming some PRs and doing equivalent changes this summer for that

echoix replied:
```suggestion
                    "Module {task_name} in your model contains "
                    "parameterized flags. actinia does not support "
                    "parameterized flags. The following flags are therefore "
                    "not being written in the generated JSON: {flags}"
                ).format(task_name=task.get_name(), flags=itemParameterizedFlags),
                caption=_("Warning"),
```

echoix commented:
Clangformat has some activity on this error today

echoix commented:
See https://github.com/DoozyX/clang-format-lint-action/pull/75#pullrequestreview-2280796910 for the second PR supposed to fix that

wenzeslaus commented on gui/wxpython/gmodeler/model.py:
```diff
@@ -2818,12 +2818,11 @@ def _getPythonActionCmd(self, item, task, cmdIndent, variables={}):
-                    f"{itemParameterizedFlags}"
-                ),
+                    "Module {task_name} in your model contains "
+                    "parameterized flags. actinia does not support "
```
I seems to use lowercase in the name, but uppercase when the context requires uppercase, especially if it is beginning of a sentence. Use uppercase here.

https://actinia-org.github.io/actinia-core/
https://www.osgeo.org/projects/actinia/

wenzeslaus commented on python/grass/temporal/univar_statistics.py:
```diff
@@ -65,12 +65,9 @@ def compute_univar_stats(registered_map_info, stats_module, fs, rast_region=Fals
-                    rmap=id, voxel="" if stats_module.name == "r.univar" else "3d "
-                )
-            )
+            _("Unable to get statistics for raster map <%s>") % id
```
Why did you choose % and %s for formatting? Because of string on line 105?

echoix replied:
To make existing translations in 4 languages (but now not collected in the templates anymore) match up again. See the long comment analyzing that that you replied to.

wenzeslaus commented on python/grass/temporal/univar_statistics.py:
```diff
@@ -65,12 +65,9 @@ def compute_univar_stats(registered_map_info, stats_module, fs, rast_region=Fals
-            )
+            _("Unable to get statistics for raster map <%s>") % id
+            if stats_module.name == "r.univar"
+            else _("Unable to get statistics for 3d raster map <%s>") % id
```
We use uppercase D in 3D in sentences, e.g., 3D view in GUI.

```suggestion
            else _("Unable to get statistics for 3D raster map <%s>") % id
```

echoix replied:
Can we do that correction after this PR is merged and translation files picked up the changes? The source string (that wasn't found anymore in English, but exists in 4 translated languages) had lowercase 3d.

There's more of my original improvements to go back after that anyways.

wenzeslaus replied:
If you think it's better that way, sure!

echoix commented:
@wenzeslaus can you re-approve? I had to resolve a conflict in t.rast.what (a line was removed)

echoix commented:
So now, we need to regenerate the template files. Last time it was @HuidaeCho that did it. Is it necessary for it to be a manual job?

petrasovaa commented:
@echoix This breaks WMS, specifically the change in `gui/wxpython/web_services/dialogs.py` change:
```
        self.statusbar.SetStatusText(
            _("Connecting to <$s>...") % self.server.GetValue().strip()
        )
```
```
Traceback (most recent call last):
  File "/home/anna/dev/grass/grass/dist.x86_64-pc-linux-
gnu/gui/wxpython/web_services/dialogs.py", line 379, in
OnConnect

_("Connecting to <$s>...") % self.server.GetValue().strip()
TypeError
:
not all arguments converted during string formatting

```

echoix commented:
Indeed, I see an unexpected `$s` instead of `%s`

echoix commented:
Do you mind filing in the PR?

petrasovaa commented:
> Do you mind filing in the PR?

@arohanajit Can you please do it? Thanks!

echoix commented:
Maybe a spot check that $s isn't anywhere else in code strings. I know there's some rules for unused arguments, maybe Pylint, but probably not active yet

echoix commented:
> > Do you mind filing in the PR?
> 
> 
> 
> @arohanajit Can you please do it? Thanks!

Did it
